docs/source/link_types/index.rst

```diff
@@ -0,0 +1,69 @@
+Node classes
+-----------
+There are two *Node* sub-classes, *Data* and *Calculation*. The *Code* nodes
+can be considered as sub-classes of *Data* nodes. *JobCalculations*,
+*InlineCalulations* and the *WorkCalculations* are subclasses of the
+*Calculation* class.
+
+AiIDA graph
+-----------
+The AiIDA graph is a `directed graph <https://en.wikipedia.org/wiki/Directed_graph>`_.
+The links of the graph are arrows connecting two nodes. An arrow (x, y) is
+considered to be directed from x to y; y is called the **head** and x is called
+the **tail** of the arrow; y is said to be a **direct successor** of x and x
+is said to be a **direct predecessor** of y. If a path leads from x to y, then
```

Comment:
I'm not sure if we want to add the discussion on direct successor or predecessor, it is not needed for this page, and I think it introduces a new "language" that we never use elsewhere...

Reply:
It's true. I had the same feeling and I mainly added it because I need it for the discussion on the export that will follow (and it would be late to be mentioned at that point).

As a general comment, I would say that it would be good to follow (or start following?) the terminology of the domain that we focus on. E.g. for the moment we use input/output and ancestor/descendant that seemed natural to us (at least this is my understanding). For trees, which are DAGS, it is used parent/child & ancestor/descendant to refer to the same relationships. At https://en.wikipedia.org/wiki/Directed_graph where they talk about directed graphs (which what we have now) they talk about (direct/indirect) predecessor /successors.